app/models/site_setting.rb

```diff
@@ -101,22 +101,24 @@ def self.email_polling_enabled?
     :markdown_typographer_quotation_marks
   ]
 
+  def self.cache
+    @cache ||= DistributedCache.new("site_settings")
+  end
+
   def self.reset_cached_settings!
-    @blocked_attachment_content_types_regex = nil
-    @blocked_attachment_filenames_regex = nil
-    @allowed_unicode_username_regex = nil
+    cache.clear
   end
 
   def self.blocked_attachment_content_types_regex
-    @blocked_attachment_content_types_regex ||= Regexp.union(SiteSetting.blocked_attachment_content_types.split("|"))
+    cache["blocked_attachment_content_types_regex"] ||= Regexp.union(SiteSetting.blocked_attachment_content_types.split("|"))
```

Comment:
Sorry on mobile here but we might want to use defer_get_set since there is no need to publish the changes immediately